web/html/src/manager/maintenance/maintenance-windows-details.js

```diff
@@ -0,0 +1,178 @@
+                        <tr>
+                            <td>{t("Calendar Name")}</td>
+                            <td>{t(data.calendarName)}</td>
+                        </tr>
+                        <tr>
+                            <td>{t("Used by schedule")}:</td>
+                            <td>{t(data.scheduleNames.map(name => name.name).join(", "))}</td>
+                        </tr>
+                        {data.calendarUrl &&
+                        <tr>
+                            <td>{t("Url")}:</td>
+                            <td>{t(data.calendarUrl)}</td>
+                        </tr>
+                        }
+                        </tbody>
+                    </table>
+                </div>
+                <DeleteDialog id="delete-modal"
+                              title={t("Delete maintenance calendar")}
+                              content={
+                                  <Form model={{strategy: this.state.strategy}} onChange={this.setCheck}>
+                                      <div>{t("Are you sure you want to delete the selected item?")}</div>
+                                      <div>{t("Deleting this calendar will unassign all schedules from this calendar.")}</div>
+                                      <Check name="strategy" label={<b>{t("Cancel affected actions?")}</b>} divClass="col-md-6" />
+                                  </Form>
+                              }
+                              onConfirm={() => this.props.onDelete(this.addStrategy())}
+                />
+            </BootstrapPanel>
+        );
+    }
+
+    render() {
+        const buttons = [
+            <div className="btn-group pull-right">
+                <Button
+                    text={t("Back")}
+                    icon="fa-chevron-left"
+                    title={t("Back")}
+                    className="btn-default"
+                    handler={() => this.props.onCancel("back")}
+                />
+                <Button
+                    text={t("Edit")}
+                    icon="fa-edit"
+                    title={t("Edit")}
+                    className="btn-default"
+                    handler={() => this.props.onEdit(
+                        this.state.type === "schedule" ? this.props.data.scheduleId : this.props.data.calendarId
+                    )}
+                />
+                <ModalButton
+                    text={t("Delete")}
+                    icon="fa-trash"
+                    title={t("Delete")}
+                    target="delete-modal"
+                    className="btn-default"
+                />
+            </div>
+        ];
+
+        return (
+            <TopPanel
+                title={this.props.data.scheduleName}
+                icon="spacewalk-icon-salt"
+                helpUrl=""
+                button={buttons}
+            >
+                { this.state.messages ?
+                    <Messages items={this.state.messages}/> : null }
```

Comment:
It can stay in a single line I guess. Anyway, isn't the `Messages` component able to digest `items = null`? It would be nice to get rid of the check here.